prompttools/experiment/experiments/langchain_experiment.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Hegel AI, Inc.
+# All rights reserved.
+#
+# This source code's license can be found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from typing import Any, Dict, List
+
+from langchain.chains import LLMChain
+from langchain.chains import SequentialChain
+
+# from pydantic.main import ModelMetaclass
+
+from time import perf_counter
+import logging
+
+from prompttools.mock.mock import mock_lc_completion_fn
+
+from .experiment import Experiment
+from .error import PromptExperimentException
+
+VALID_TASKS = ()
+
+
+class SequentialChainExperiment(Experiment):
+    """
+    Experiment for LangChain sequential chains.
+    For sequential chains, the order in which
+    links in the chain are submitted to the
+    class is very strict. Users should take extra
+    precaution since this is not yet enforced.
+    """
+
+    MODEL_PARAMETERS = ["temperature"]
+    CALL_PARAMETERS = {"prompt_chain", "temperature", "input_variables", "output_variables", "output_key", "prompt"}
+
+    def __init__(
+        self,
+        llms: List[Any],
+        **kwargs: Dict[str, object],
+    ):
+        self.completion_fn = self.lc_completion_fn
+        self.prompt = kwargs["prompt"]
+        self.call_params = {k: kwargs[k] for k in self.CALL_PARAMETERS if k in self.CALL_PARAMETERS}
+        self.output_keys = kwargs["output_key"]
+        self.llms = []
+        # Only supporting temperature for now
+        for llm in llms:
```

Comment:
Is there a use case where each link in a chain is using a different LLM? Do we need to take a list of list of LLMs to support that?

Reply:
There's a dependency challenge:
LLMChain depends on the LLM to be an input -> the LLM depends on temperature to be an input:

```
llm = ChatOpenAI(temperature=0.9)

# prompt template 1: translate to english
first_prompt = ChatPromptTemplate.from_template(
    "Translate the following review to english:"
    "\n\n{Review}"
)
# chain 1: input= Review and output= English_Review
chain_one = LLMChain(llm=llm, prompt=first_prompt, 
                     output_key="English_Review"
                    )
```

So I think we can:
1. Require the user to have already defined temperature before input to the experiment (which takes away the benefit of param combos)
2. The combos are generated as separate LLMs which can then be tested